styx-common/src/test/java/com/spotify/styx/util/TimeUtilTest.java

```diff
@@ -187,53 +191,149 @@ public void shouldAddOffsetWithWeek() {
-    final Instant lastTimeMonths = Instant.parse("2016-01-01T00:00:00.00Z");
-
-    assertThat(instancesInRange(firstTimeHours, lastTimeHours, Schedule.HOURS), is(9));
-    assertThat(instancesInRange(firstTimeDays, lastTimeDays, Schedule.DAYS), is(9));
-    assertThat(instancesInRange(firstTimeWeeks, lastTimeWeeks, Schedule.WEEKS), is(1));
-    assertThat(instancesInRange(firstTimeMonths, lastTimeMonths, Schedule.YEARS), is(6));
+    final Instant firstTimeHours = parse("2016-01-19T00:00:00.00Z");
+    final Instant lastTimeHours = parse("2016-01-19T03:00:00.00Z");
+    final Instant firstTimeDays = parse("2016-01-10T00:00:00.00Z");
+    final Instant lastTimeDays = parse("2016-01-13T00:00:00.00Z");
+    final Instant firstTimeWeeks = parse("2016-01-11T00:00:00.00Z");
+    final Instant lastTimeWeeks = parse("2016-01-18T00:00:00.00Z");
+    final Instant firstTimeMonths = parse("2010-01-01T00:00:00.00Z");
+    final Instant lastTimeMonths = parse("2013-01-01T00:00:00.00Z");
+
+    assertThat(instantsInRange(firstTimeHours, lastTimeHours, Schedule.HOURS),
+        contains(parse("2016-01-19T00:00:00.00Z"),
+            parse("2016-01-19T01:00:00.00Z"),
+            parse("2016-01-19T02:00:00.00Z")));
+
+    assertThat(instantsInRange(firstTimeDays, lastTimeDays, Schedule.DAYS),
+        contains(parse("2016-01-10T00:00:00.00Z"),
+            parse("2016-01-11T00:00:00.00Z"),
+            parse("2016-01-12T00:00:00.00Z")));
+
+    assertThat(instantsInRange(firstTimeWeeks, lastTimeWeeks, Schedule.WEEKS),
+        contains(parse("2016-01-11T00:00:00.00Z")));
+
+    assertThat(instantsInRange(firstTimeMonths, lastTimeMonths, Schedule.YEARS),
+        contains(parse("2010-01-01T00:00:00.00Z"),
+            parse("2011-01-01T00:00:00.00Z"),
+            parse("2012-01-01T00:00:00.00Z")));
   }
 
   @Test
   public void shouldGetCorrectNumberOfInstantsForCron() {
-    final Instant firstTimeHours = Instant.parse("2016-01-19T00:00:00.00Z");
-    final Instant lastTimeHours = Instant.parse("2016-01-19T09:00:00.00Z");
+    final Instant firstTimeHours = parse("2016-01-19T00:00:00.00Z");
+    final Instant lastTimeHours = parse("2016-01-19T03:00:00.00Z");
 
-    assertThat(instancesInRange(firstTimeHours, lastTimeHours, Schedule.parse("0 * * * *")),
-               is(9));
+    assertThat(instantsInRange(firstTimeHours, lastTimeHours, Schedule.parse("0 * * * *")),
+        contains(parse("2016-01-19T00:00:00.00Z"),
+            parse("2016-01-19T01:00:00.00Z"),
+            parse("2016-01-19T02:00:00.00Z")));
+  }
+
+  @Test
+  public void shouldReturnEmptyList() {
+    final Instant firstTimeHours = parse("2016-01-19T00:00:00.00Z");
+    final Instant lastTimeHours = parse("2016-01-19T00:00:00.00Z");
+
+    assertTrue(
+        instantsInRange(firstTimeHours, lastTimeHours, Schedule.parse("0 * * * *")).isEmpty());
   }
 
   @Test(expected = IllegalArgumentException.class)
   public void shouldGetExceptionIfLastInstantIsBeforeFirstInstant() {
```

Comment:
Maybe not part of this PR but this use of `@Test(expected = IllegalArgumentException.class)` might produce false positives for these tests if `parse` throws.